docs/topics/whatsnew1720.md

```diff
@@ -0,0 +1,749 @@
+  and files when traversing them.
+* `visitFileTree(fileVisitor: FileVisitor, ...)` consumes a ready `FileVisitor` and uses `java.nio.file.Files.walkFileTree()`
+  under the hood.
+* `visitFileTree(..., builderAction: FileVisitorBuilder.() -> Unit)` creates a `FileVisitor` with the `builderAction` and
+  calls the `visitFileTree(fileVisitor, ...)` function.
+* `FileVisitResult`, return type of `FileVisitor`, has the `CONTINUE` default value that continues the processing of the
+  file.
+
+> The new extension functions for `java.nio.file.Path` are [Experimental](components-stability.md).
+> They may be changed at any time. Opt-in is required (see details below), and you should use them only for evaluation purposes.
+>
+{type="warning"}
+
+Here are some things you can do with these new extension functions:
+
+* Explicitly create a `FileVisitor` and then use:
+
+  ```kotlin
+  val cleanVisitor = fileVisitor {
+      onPreVisitDirectory { directory, attributes ->
+          // Some logic on visiting directories
+          FileVisitResult.CONTINUE
+      }
+  
+      onVisitFile { file, attributes ->
+          // Some logic on visiting files
+          FileVisitResult.CONTINUE
+      }
+  }
+  
+  // Some logic may go here
+  
+  projectDirectory.visitFileTree(cleanVisitor)
+  ```
+
+* Create a `FileVisitor` with the `builderAction` and use it immediately:
+
+  ```kotlin
+  projectDirectory.visitFileTree {
+  // Definition of the poAction:
+      onPreVisitDirectory { directory, attributes ->
+          // Some logic on visiting directories
+          FileVisitResult.CONTINUE
+      }
+  
+      onVisitFile { file, attributes ->
+          // Some logic on visiting files
+          FileVisitResult.CONTINUE
+      }
+  }
+  ```
+
+* Traverse a file tree rooted at the specified path with the `walk()` function:
+
+  ```kotlin
+  @OptIn(kotlin.io.path.ExperimentalPathApi::class)
+  fun taverseFileTree() {
+      val cleanVisitor = fileVisitor {
+          onPreVisitDirectory { directory, _ ->
+              if (directory.name == "build") {
+                  directory.toFile().deleteRecursively()
+                  FileVisitResult.SKIP_SUBTREE
+              } else {
+                  FileVisitResult.CONTINUE
+              }
+          }
+  
+          onVisitFile { file, _ ->
+              if (file.extension == "class") {
+                  file.deleteExisting()
+              }
+              FileVisitResult.CONTINUE
+          }
+      }
+  
+      val rootDirectory = createTempDirectory("Project")
+  
+      rootDirectory.resolve("src").let { srcDirectory ->
+          srcDirectory.createDirectory()
+          srcDirectory.resolve("A.kt").createFile()
+          srcDirectory.resolve("A.class").createFile()
+      }
+  
+      rootDirectory.resolve("build").let { buildDirectory ->
+          buildDirectory.createDirectory()
+          buildDirectory.resolve("Project.jar").createFile()
+      }
+  
+   
+  //sampleStart
```

Comment:
For some reason another code is shown too; maybe it doesn't work without `main`, need to check.